Feature/save result

jonjohansen opened the pull request:
## Changes :balloon:
#### Images can now be saved! 🌆 
## Types of changes
### Results can now be saved for iOS and Droid!
* A button has been added to save (and in future share) the results of your diploma.
* An interface that is dependent on OS has been made.
* The android project has been renamed to Droid to avoid conflicting namespaces :^)

What types of changes does your code introduce? _Mark boxes that apply with an `x`._

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist :heavy_check_mark:

_Mark boxes that apply with an `x`. These can also be filled out after the pull request is created._

- [ ] I have read the coding convention [guidelines](https://github.com/Dualog-students/dualog-students.github.io#company-guidelines) and my code follows these.
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

<!-- The structure of your pull request can be altered if the template does not suit your needs -->

jonjohansen committed:
Add button to navigation menu

jonjohansen committed:
Saving images!

Interface for saving images has been added
 - Implemented on iOS
 - Permissions have been set for iOS

Views:
Result has gotten two buttons
Save button now has an action, which should run a code-behind function for saving
 NOTE: the name of the image saved in the codebehind should be changed

jonjohansen committed:
Image save!

It now works on android!
Filename sent in is "Navn" for now. This needs to change

IsakSundeSingh commented on Diplomatic.Android/Classes/Picture_Droid.cs:
```diff
@@ -0,0 +1,35 @@
+        public void SavePictureToDisk(string filename, byte[] imageArray)
+        {
+            var dir = Android.OS.Environment.GetExternalStoragePublicDirectory(Android.OS.Environment.DirectoryDcim);
+            var pictures = dir.Path;
```
Use explicit types, man.

IsakSundeSingh commented on Diplomatic.Android/Classes/Picture_Droid.cs:
```diff
@@ -0,0 +1,35 @@
+            {
+                // Write file to the android disk
+                System.IO.File.WriteAllBytes(FilePath, imageArray);
+                // Now it needs to be added to image gallery
```
While you're at it, add a newline between 23 and 24. ✨

IsakSundeSingh commented on Diplomatic.Android/Classes/Picture_Droid.cs:
```diff
@@ -0,0 +1,35 @@
+                mediaScanIntent.SetData(Android.Net.Uri.FromFile(new File(FilePath)));
+                Xamarin.Forms.Forms.Context.SendBroadcast(mediaScanIntent);
+            }
+            catch(System.Exception e)
```
Space after catch-keyword. Basically `ctrl-k-d`

IsakSundeSingh commented on Diplomatic.iOS/Classes/Picture_iOS.cs:
```diff
@@ -0,0 +1,26 @@
+{
+    public class Picture_iOS : IPicture
+    {
+        
```
Unnecessary newline

IsakSundeSingh commented on Diplomatic.iOS/Classes/Picture_iOS.cs:
```diff
@@ -0,0 +1,26 @@
+            {
+                if (error != null)
+                {
+                    Console.WriteLine(error);
```
Don't need to both log it and throw it.

IsakSundeSingh commented on Diplomatic.iOS/Info.plist:
```diff
@@ -21,7 +21,7 @@
 	</array>
 	<key>MinimumOSVersion</key>
-	<string>8.0</string>
+	<string>11.3</string>
```
Should we have the newest OS version as the minimum target version?

IsakSundeSingh commented on Diplomatic/ViewModels/ResultViewModel.cs:
```diff
@@ -8,6 +8,12 @@ namespace Diplomatic.ViewModels
+        {
+            ImageUri = imageuri;
+            Filename = filename;
+            
```
Unnecessary newline.

IsakSundeSingh commented on Diplomatic/Views/Result.xaml.cs:
```diff
@@ -1,20 +1,29 @@
+                byte[] imageBytes = await webClient.DownloadDataTaskAsync(((ResultViewModel)BindingContext).ImageUri);
+                DependencyService.Get<IPicture>().SavePictureToDisk(Filename, imageBytes);
+                await DisplayAlert("Save diploma", "Your diploma has been saved!", "Ok");
+
```
Unnecessary newline.

IsakSundeSingh commented:
Nice, but the last commit is a fixup commit.

jonjohansen committed:
Ground breaking new feature

Images now have names (Only available on an ANDROID near you)!
Also
	Stuff looks clean now

IsakSundeSingh reviewed (approved):
Aight, I guess...
